components/snowflake/snowflake.app.mjs

```diff
@@ -98,6 +99,22 @@ export default {
       await promisify(this.connection.connect).bind(this.connection)();
       return this.connection;
     },
+    _extractPrivateKey(key, passphrase) {
+      if (!key.startsWith("-----BEGIN ENCRYPTED PRIVATE KEY-----")) {
+        // Key is not encrypted, no need to extract anything
+        return key;
+      }
+
+      const privateKeyObject = createPrivateKey({
+        key,
+        format: "pem",
+        passphrase,
+      });
+      return privateKeyObject.export({
+        format: "pem",
+        type: "pkcs8",
+      });
+    },
```

Comment:
For more context, see this issue:
https://github.com/snowflakedb/snowflake-connector-nodejs/issues/855